armeria-backend/src/main/scala/sttp/client3/armeria/ArmeriaBackend.scala

```diff
@@ -0,0 +1,198 @@
+package sttp.client3.armeria
+
+import com.linecorp.armeria.client.{ResponseTimeoutException, WebClient}
+import com.linecorp.armeria.common.MediaType.{OCTET_STREAM, PLAIN_TEXT}
+import com.linecorp.armeria.common.{
+  AggregatedHttpResponse,
+  ClosedSessionException,
+  HttpMethod,
+  HttpRequest,
+  RequestHeaders,
+  ResponseHeaders
+}
+import io.netty.util.AsciiString
+import sttp.capabilities
+import sttp.client3.SttpClientException.ReadException
+import sttp.client3.armeria.ArmeriaBackend.{
+  MultipartNotSupportedException,
+  PseudoHeaderPrefix,
+  StreamingNotSupportedException
+}
+import sttp.client3.internal.{BodyFromResponseAs, FileHelpers, SttpFile}
+import sttp.client3.ws.{GotAWebSocketException, NotAWebSocketException}
+import sttp.client3.{
+  ByteArrayBody,
+  ByteBufferBody,
+  FileBody,
+  FollowRedirectsBackend,
+  InputStreamBody,
+  MultipartBody,
+  NoBody,
+  Request,
+  Response,
+  StreamBody,
+  StringBody,
+  SttpBackend,
+  SttpClientException,
+  WebSocketResponseAs
+}
+import sttp.model._
+import sttp.monad.{FutureMonad, MonadError}
+
+import java.io._
+import java.nio.charset.StandardCharsets.UTF_8
+import java.time.Duration
+import java.util
+import java.util.Map.Entry
+import java.util.zip.{GZIPInputStream, InflaterInputStream}
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future}
+import scala.io.Source
+import scala.jdk.FutureConverters._
+import scala.util.Try
+
+class ArmeriaBackend(implicit ec: ExecutionContext = ExecutionContext.global) extends SttpBackend[Future, Any] {
+  type PE = Any with capabilities.Effect[Future]
+
+  override def send[T, R >: PE](request: Request[T, R]): Future[Response[T]] =
+    adjustExceptions(request)(execute(request))
+
+  private def execute[T, R >: PE](request: Request[T, R]): Future[Response[T]] =
+    WebClient
+      .builder()
+      .responseTimeout(Duration.ofMillis(request.options.readTimeout.toMillis))
+      .build()
```

Comment:
shouldn't we build the `WebClient` once and re-use it for each request?

We might in fact parametrise the backend with the client and allow the user to provide a custom one (in the factory methods)

Reply:
good point

Reply:
I parameterized backend with client, but when it's not provided we have to build a new one since it's immutable and we want to adjust timeout based on options from our domain Request object